packages/ui/src/components/dropdown-menu.tsx

```diff
@@ -1,199 +1,438 @@
+            <div className="cn-dropdown-menu-container">{children}</div>
+          </ScrollArea>
+
+          {!!footer && <div className="cn-dropdown-menu-container cn-dropdown-menu-container-footer">{footer}</div>}
+        </Primitive>
+      </DropdownMenuPortal>
+    )
   }
->(({ className, inset, children, ...props }, ref) => (
-  <DropdownMenuPrimitive.SubTrigger
-    ref={ref}
-    className={cn(
-      'focus:bg-cn-background-3 focus:outline-none data-[state=open]:bg-cn-background-3 flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none',
-      inset && 'pl-8',
-      className
+)
+DropdownMenuContent.displayName = displayNames.content
+
+const DropdownMenuGroup = forwardRef<
+  ElementRef<typeof DropdownMenuPrimitive.Group>,
+  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Group> & { label?: string }
+>(({ children, label, ...props }, ref) => (
+  <DropdownMenuPrimitive.Group ref={ref} {...props}>
+    {label && (
+      <DropdownMenuPrimitive.Label className="cn-dropdown-menu-group-label">{label}</DropdownMenuPrimitive.Label>
     )}
-    {...props}
-  >
     {children}
-    <ChevronRightIcon className="ml-auto size-4" />
-  </DropdownMenuPrimitive.SubTrigger>
+  </DropdownMenuPrimitive.Group>
 ))
-DropdownMenuSubTrigger.displayName = DropdownMenuPrimitive.SubTrigger.displayName
+DropdownMenuGroup.displayName = displayNames.group
 
-const DropdownMenuSubContent = React.forwardRef<
-  React.ElementRef<typeof DropdownMenuPrimitive.SubContent>,
-  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubContent>
->(({ className, ...props }, ref) => (
-  <DropdownMenuPrimitive.SubContent
-    ref={ref}
-    className={cn(
-      'bg-cn-background-3 text-cn-foreground-1 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 min-w-[8rem] overflow-hidden rounded-md border p-1 shadow-lg',
-      className
+interface DropdownBaseItemProps {
+  title: ReactNode
+  children?: ReactNode
+  className?: string
+  description?: ReactNode
+  label?: string
+  shortcut?: string
+  checkmark?: boolean
+  withSubIndicator?: ReactNode
+}
+
+const DropdownBaseItem = ({
+  children,
+  className,
+  title,
+  description,
+  label,
+  shortcut,
+  checkmark,
+  withSubIndicator
+}: DropdownBaseItemProps) => (
+  <div className={cn('cn-dropdown-menu-base-item', className)}>
+    {children}
+    <div className="grid w-full">
+      {typeof title === 'string' ? (
+        <Text className="grid" color="foreground-1">
+          {title}
+        </Text>
+      ) : (
+        title
+      )}
+      {typeof description === 'string' ? <Text color="foreground-2">{description}</Text> : description}
```

Comment:
same here.. let's use a Text variant

Reply:
Answered here - https://github.com/harness/canary/pull/1673#discussion_r2126448645

Also removed color prop, since `foreground-2` is a default value